test/error-messages/nested-invalid-value.txt

```diff
@@ -0,0 +1,9 @@
+Invalid value
+
+concat :: Array a -> Array a -> Array a
+                     ^^^^^^^
+                        1
+
+1)  "ab" :: String
+
+The value at position 1 is not a member of (Array Number).
```

Comment:
This change is not crucial. Unless there's a straightforward solution, let's not worry about it.